packages/api/tsserver/tsserver.mts

```diff
@@ -0,0 +1,134 @@
+import { parseTsServerMessages } from './utils.mjs';
+import type { ChildProcess } from 'child_process';
+import type { server as tsserver } from 'typescript';
+
+export class TsServer {
+  private _seq: number = 0;
+  private readonly process: ChildProcess;
+  private readonly resolvers: Record<number, (value: any) => void> = {};
+
+  constructor(process: ChildProcess) {
+    this.process = process;
+    this.process.stdout?.on('data', (buffer) => {
+      for (const message of parseTsServerMessages(buffer)) {
+        if (message.type === 'response') {
+          this.handleResponse(message);
+        } else if (message.type === 'event') {
+          this.handleEvent(message);
+        }
+      }
+    });
+  }
+
+  private get seq() {
+    return this._seq++;
+  }
+
+  private handleResponse(response: tsserver.protocol.Response) {
+    const resolve = this.resolvers[response.request_seq];
+
+    if (!resolve) {
+      console.warn(
+        `Received a response for command '${response.command}' and request_seq '${response.request_seq}' but no resolver was found. Did you forget to register a resolver?`,
```

Comment:
Not a problem, but this is funny because you say "You" talking to the dev. Usually a passive form like "was a resolver set?" is used

Reply:
Meant to enable this only for development so it doesn't appear on users machines. I just changed it though as we don't currently have the process.env set for production in the API.